src/overlays/actors/ovl_Bg_Spot18_Basket/z_bg_spot18_basket.c

```diff
@@ -21,39 +36,397 @@ const ActorInit Bg_Spot18_Basket_InitVars = {
+        this->unk_218 = 1;
+    } else {
+        this->unk_218 = 0;
+    }
+
+    this->unk_216 = 0;
+}
+
+void func_808B7FC0(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 tempUnk214;
+    f32 tempUnk210;
+    s16 arrayValue;
+    f32 clampedTempUnk210;
+
+    this->unk_212 += 0xBB8;
+
+    if (this->unk_216 >= 13) {
+        tempUnk214 = Math_ApproxS(&this->unk_214, 0, 55);
+    } else {
+        tempUnk214 = 0;
+    }
+
+    this->dyna.actor.shape.rot.x = Math_Coss(this->unk_212) * this->unk_214;
+    this->dyna.actor.shape.rot.z = -Math_Sins(this->unk_212) * this->unk_214;
+
+    Math_ApproxS(&this->unk_210, 500, 10);
+    this->dyna.actor.shape.rot.y += this->unk_210;
+
+    if (tempUnk214 != 0) {
+        arrayValue = D_808B85C8[this->unk_218];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/BgSpot18Basket_Init.s")
+        if ((s16)(this->dyna.actor.shape.rot.y - arrayValue) >= 0) {
+            this->dyna.actor.shape.rot.y = arrayValue;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/BgSpot18Basket_Destroy.s")
+            func_808B818C(this);
+            func_8003EC50(globalCtx, &globalCtx->colCtx.dyna, this->dyna.dynaPolyId);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7AEC.s")
+    if (this->unk_216 < 30) {
+        func_808B7770(this, globalCtx, 0.5f);
+    } else {
+        func_808B7770(this, globalCtx, 0.3f);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7AFC.s")
+    tempUnk210 = (this->unk_210 - 500) * 0.0006f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7B58.s")
+    clampedTempUnk210 = CLAMP(tempUnk210, 0.0f, 1.5f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7B6C.s")
+    func_800F436C(&this->dyna.actor.projectedPos, 203, clampedTempUnk210);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7BB0.s")
+void func_808B818C(BgSpot18Basket* this) {
+    this->actionFunc = func_808B81A0;
+    this->unk_216 = 0;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7BCC.s")
+void func_808B81A0(BgSpot18Basket* this, GlobalContext* globalCtx) {
+    s32 i;
+    Actor* actor = &this->dyna.actor;
+    Vec3f tempVector;
+    EnItem00* collectible;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7D38.s")
+    if (this->unk_216 == 1) {
+        tempVector.x = actor->posRot.pos.x;
+        tempVector.y = actor->posRot.pos.y + 170.0f;
+        tempVector.z = actor->posRot.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Spot18_Basket/func_808B7D50.s")
+        if (this->unk_218 == 0) {
+            for (i = 0; i < ARRAY_COUNT(D_808B85E4); i++) {
+                collectible = Item_DropCollectible(globalCtx, &tempVector, 4);
```

Comment:
You can use the `Item00Type` enum to give some sense to all these `Item_DropCollectible` calls
```suggestion
                collectible = Item_DropCollectible(globalCtx, &tempVector, ITEM00_BOMBS_A);
```